pkg/capabilities/mercury/mercury.go

```diff
@@ -54,27 +54,33 @@ func FeedIDFromBytes(b [FeedIDBytesLen]byte) FeedID {
 }
 
 type FeedReport struct {
-	FeedID               string `json:"feedId"`
-	FullReport           []byte `json:"fullReport"`
-	BenchmarkPrice       int64  `json:"benchmarkPrice"`
-	ObservationTimestamp int64  `json:"observationTimestamp"`
+	FeedID     string
+	FullReport []byte
+	Rs         [][]byte
+	Ss         [][]byte
+	Vs         []byte
+
+	// Fields below are derived from FullReport
+	// NOTE: BenchmarkPrice is a byte representation of big.Int. We can't use big.Int
+	// directly due to Value serialization problems using mapstructure.
+	BenchmarkPrice       []byte
+	ObservationTimestamp int64
 }
 
-type Codec struct {
+// Codec that doesn't validate signatures.
```

Comment:
Are Codecs meant to validate signatures more generally? Why does this not validate signatures?